### ISSUE TITLE
add a field to KubernetesDiscovery to ignore child pods

### ISSUE THREAD

**Issue by nicks** [tilt-dev/tilt#4867]
Currently, the KubernetesDiscovery API has two mechanism for tracking child pods:

https://api.tilt.dev/kubernetes/kubernetes-discovery-v1alpha1.html

1) Watch all the pods in the namespace, and traverse owner references to the UID of the object deployed.
2) An extra_pod_selectors to select pods by label

(1) is much more automatic but is sometimes wrong. For example: you might have a CRD that does some of its work in pods, then creates a new Deployment. In that case, the child pods of the CRD aren't the ones we want to track for readiness purposes.  You want the ones from the deployment.

We should have some mechanism to ignore the child pods.

**Comment by nicks** [comment]
I _could_ imagine an API where, if ExtraPodSelectors is specified, then we don't track child pods. I go back and forth on whether this would be confusing or helpful.

**Comment by mingxiao** [comment]
Perhaps you could have an ExcludePodSelectors such that during owner traversal, any matching pods are excluded?

**Comment by nicks** [comment]
@mingxiao ya, thinking of a syntax like: 

```
k8s_resource(..., extra_pod_selectors=[{...}], ignore_owner_references=True)
```

**Comment by lisamburns** [comment]
My two cents... the owner references seems like a useful flag. As a separate issue, I was a little surprised that Tilt was tracking these pods given that they had already exited (status was Completed).

**Comment by nicks** [comment]
thanks @lisamburns ! Ya, tilt currently tracks completed pods so that, for pods that exit quickly, the log streamer will still get the logs. This PR fixes the live-update part so that it filters out exited pods: https://github.com/tilt-dev/tilt/pull/4873. So that should mostly unblock you.

**Comment by nicks** [comment]
One side note on this - the naive way to implement this would be to skip owner references when discovering pods. Live update won't attach to the pods. You won't get portforwards. You also won't get logs.

I wonder if, in the future, we will need to have a notion of a resource that owns different types of pods, so that you get logs from the builder pods but only do live-update on the deploy pods.

**Comment by nicks** [comment]
ended up implementing this as

```
k8s_resource(discovery_mode='selectors-only')
```

which will be in the next tilt release

**Comment by paulcwarren** [comment]
> I wonder if, in the future, we will need to have a notion of a resource that owns different types of pods, so that you get logs from the builder pods but only do live-update on the deploy pods.

For what it is worth I can definitely see this use case Nick.  Essentially, a tiltfile dsl allowing for the creation and grouping podlogstreams.  Separate from pods that are the target of live-update.
